engine/Assets/Scripts/Importer/Importer.cs

```diff
@@ -257,9 +259,16 @@ public static (GameObject MainObject, MirabufLive miraLive, SimObject Sim)
                         // TODO: Implement and test limits
                         var limits = definition.Rotational.RotationalFreedom.Limits;
                         if (limits != null && limits.Lower != limits.Upper) {
+                            var currentPosition = definition.Rotational.RotationalFreedom.Value;
+                            while (currentPosition < -Mathf.PI) currentPosition += 2 * Mathf.PI;
+                            while (currentPosition > Mathf.PI) currentPosition -= 2 * Mathf.PI;
+                            var min = -(limits.Upper - currentPosition);
+                            while (min > 0) min -= 2 * Mathf.PI;
+                            var max = -(limits.Lower - currentPosition);
+                            while (max < 0) max += 2 * Mathf.PI;
                             revoluteA.useLimits = true;
-                            revoluteA.limits    = new JointLimits() { min = -limits.Upper * Mathf.Rad2Deg,
-                                   max                                    = -limits.Lower * Mathf.Rad2Deg };
+                            revoluteA.limits    = new JointLimits() { min = min * Mathf.Rad2Deg, max = max * Mathf.Rad2Deg };
+                            revoluteA.extendedLimits = true;
                         }
```

Comment:
```suggestion
                        if (limits != null && limits.Lower != limits.Upper) {
                            var currentPosition = definition.Rotational.RotationalFreedom.Value;
                            currentPosition = (currentPosition + Mathf.PI) % (2 * Mathf.PI) - Mathf.PI;

                            var min = -(limits.Upper - currentPosition);
                            min = (min % (2 * Mathf.PI) + 2 * Mathf.PI) % (2 * Mathf.PI) - 2 * Mathf.PI;

                            var max = -(limits.Lower - currentPosition);
                            max = (max % (2 * Mathf.PI) + 2 * Mathf.PI) % (2 * Mathf.PI);

                            revoluteA.useLimits = true;
                            revoluteA.limits    = new JointLimits() { min = min * Mathf.Rad2Deg, max = max * Mathf.Rad2Deg };
                            revoluteA.extendedLimits = true;
                        }
```

I'm pretty sure that this will create the same output but someone should double check me on that.

Reply:
Don't see how this works for when currentPos < -pi. I think min and max work, but wouldn't min = (min % (2 * Mathf.PI) - 2 * Mathf.PI) % (2 * Mathf.PI) be more efficient? Good idea avoiding loops.

Reply:
That would not put `min` in the range of `[-2PI, 0]`, you can check this by running python code against a [Text Field Calculator](https://samfundev.github.io/TextFieldCalculator/Calculator.html). Although now that I think about it is that even the goal here? I looked at `currentPosition` and assumed the range that was wanted by `max` and `min`, however I'm not sure what the goal is now that I look at it again.

Reply:
Given that `%` creates a value between `[-k, k]` instead of `[0, k]` (like how it should). You are correct and we need to change this. I just updated my suggestion. Make sure it actually compiles before you merge into `dev`. Credit goes to @a-crowell for the one-liner for bounding `currentPosition`.